File: backend/src/main/java/com/snackoverflow/toolgether/domain/postimage/repository/PostImageRepository.java
package com.snackoverflow.toolgether.domain.postimage.repository;

import com.snackoverflow.toolgether.domain.postimage.entity.PostImage;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PostImageRepository extends JpaRepository<PostImage, Long> {

    List<PostImage> findByPostId(Long postId);
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/postimage/repository/PostImageRepository.kt
package com.snackoverflow.toolgether.domain.postimage.repository

import com.snackoverflow.toolgether.domain.postimage.entity.PostImage
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface PostImageRepository : JpaRepository<PostImage, Long> {
    fun findAllByPostId(postId: Long): List<PostImage> // 특정 게시물의 이미지 리스트 조회

    fun findByPostId(postId: Long): List<PostImage>
    fun deleteByPostId(postId: Long)

    @Query("SELECT pi.imageUrl FROM PostImage pi")
    fun findAllImageUrl(): List<String>
}
